compiler/src/dmd/errors.d

```diff
@@ -37,6 +37,23 @@ enum ErrorKind
     message,
 }
 
+/**
+Represents a diagnostic message generated during compilation, such as errors, warnings, or other messages.
+
+Params:
+  loc = The location in the source code where the diagnostic was generated (includes file, line, and column).
+  message = The text of the diagnostic message, describing the issue.
+  kind = The type of diagnostic, indicating whether it is an error, warning, deprecation, etc.
+*/
+struct Diagnostic
+{
+    SourceLoc loc;
+    string message;
+    ErrorKind kind;
+}
+
+Diagnostic[] diagnostics; // Global array to store diagnostics
```

Comment:
this is thread local, which it shouldn't be. make it `__gshared`.  also it should be set to empty with https://github.com/dlang/dmd/blob/master/compiler/src/dmd/frontend.d#L172